src/DynamoCore/Graph/Workspaces/LayoutExtensions.cs

```diff
@@ -426,6 +432,52 @@ private static void SaveLayoutGraph(this WorkspaceModel workspace, List<GraphLay
                     node.ReportPosition();
                     workspace.HasUnsavedChanges = true;
 
+                    double noteOffset = -n.NotesHeight;
+                    foreach (NoteModel note in n.LinkedNotes)
+                    {
+                        if (note.IsSelected || DynamoSelection.Instance.Selection.Count == 0)
+                        {
+                            note.X = node.X;
+                            note.Y = node.Y + noteOffset;
+                            noteOffset += note.Height + GraphLayout.Graph.VerticalNoteDistance;
+                            note.ReportPosition();
+                        }
+                    }
+                }
+            }
+        }
+        /// <summary>
+        /// This method pushes changes from the GraphLayout.Graph objects
+        /// back to the workspace models.
+        /// </summary>
+        private static void SaveLayoutGraphForNodeAutoComplete(this WorkspaceModel workspace, List<GraphLayout.Graph> layoutSubgraphs, Guid? originalNodeGUID)
+        {
+            var originalNode = workspace.Nodes.FirstOrDefault(n => n.GUID == originalNodeGUID);
+            GraphLayout.Graph originalNodegraph = layoutSubgraphs
+                    .FirstOrDefault(g => g.FindNode(originalNode.GUID) != null);
+            GraphLayout.Node ogn = originalNodegraph.FindNode(originalNode.GUID);
+            double marginX = originalNode.X - ogn.X;
+            double marginY = originalNode.Y - ogn.Y;
+
+            // Assign coordinates to nodes outside groups
+            foreach (var node in workspace.Nodes)
+            {
+                GraphLayout.Graph graph = layoutSubgraphs
+                    .FirstOrDefault(g => g.FindNode(node.GUID) != null);
+
+                if (graph != null)
+                {
+                    GraphLayout.Node n = graph.FindNode(node.GUID);
+                    double offsetY = graph.OffsetY;
+                    if (node.GUID != originalNodeGUID )
```

Comment:
@aparajit-pratap here original node is skipped, so coordinates remain unchanged, thereby preventing node from jumping